test/e2e/utils/grafana/client.go

```diff
@@ -0,0 +1,83 @@
+package grafana
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/url"
+
+	"github.com/go-openapi/strfmt"
+	grafanaclient "github.com/grafana/grafana-openapi-client-go/client"
+	grafanasearch "github.com/grafana/grafana-openapi-client-go/client/search"
+	"github.com/scylladb/scylla-operator/pkg/pointer"
+)
+
+type Client struct {
+	c *grafanaclient.GrafanaHTTPAPI
+}
+
+type ClientOptions struct {
+	URL      string
+	Username string
+	Password string
+	TLS      *tls.Config
+}
+
+func NewClient(opts ClientOptions) (*Client, error) {
+	u, err := url.Parse(opts.URL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse url: %w", err)
+	}
+
+	cfg := grafanaclient.TransportConfig{
+		Host:      u.Host,
+		BasePath:  "/api",
+		Schemes:   []string{u.Scheme},
+		BasicAuth: url.UserPassword(opts.Username, opts.Password),
+		TLSConfig: opts.TLS,
+	}
+
+	c := grafanaclient.NewHTTPClientWithConfig(strfmt.Default, &cfg)
+
+	return &Client{
+		c: c,
+	}, nil
+}
+
+func (c *Client) Dashboards() ([]Dashboard, error) {
+	searchParams := grafanasearch.NewSearchParams()
+	searchParams.Limit = pointer.Ptr[int64](1000)
+	searchParams.Type = pointer.Ptr("dash-db")
+
+	resp, err := c.c.Search.Search(searchParams)
```

Comment:
shouldn't this be rerun with `searchParams.Page` set? if you hit the limit (`len(resp.GetPayload()) >= limit`)?